File: RedSheet.DbEntities/Models/ApplicationModule.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ApplicationModules", Schema = "dbo")]
	[Serializable]
	public partial class ApplicationModule  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ApplicationModuleId { get; set; } 

	public Nullable<int> ParentApplicationModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1)]
	public string VisibleActionItem { get; set; } 
 
	[RelationshipTableAttribue("ModuleMasters","dbo")]
	//Relationships
	public int ModuleMasterId { get; set; } 
 
	[ForeignKey("ModuleMasterId")]
	public ModuleMaster ModuleMaster { get; set; } 
	[InverseProperty("ApplicationModule")]
	public ICollection<EmailTemplateDetail> EmailTemplateDetails { get; set; } 

	[InverseProperty("ApplicationModule")]
	public ICollection<RolePermission> RolePermissions { get; set; } 

	[InverseProperty("ApplicationModule")]
	public ICollection<ModuleContent> ModuleContents { get; set; } 


	public ApplicationModule()
	{
	this.EmailTemplateDetails = new HashSet<EmailTemplateDetail>();

	this.RolePermissions = new HashSet<RolePermission>();

	this.ModuleContents = new HashSet<ModuleContent>();

    }

    }
}

File: redsheet-portal/src/app/database-models/application-module.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ModuleMaster, EmailTemplateDetail, RolePermission, ModuleContent,  } from './'
export class ApplicationModule {
    constructor(applicationModule?: ApplicationModule )  {
        let properties = [ "applicationModuleId", "parentApplicationModuleId", "visibleActionItem", "moduleMasterId", "emailTemplateDetails", "rolePermissions", "moduleContents",];
        for (let property of properties)
            if (applicationModule && applicationModule[property])
                this[property] = applicationModule[property];
    }
 
	applicationModuleId : number =   0 ;
 
	parentApplicationModuleId : number =   undefined;
 
    @required()
    @maxLength(1)
	visibleActionItem : string =   undefined;
 
    @range(0,2147483647)
	moduleMasterId : number =   undefined;
	moduleMaster : ModuleMaster  ;
	@nested(EmailTemplateDetail)
	emailTemplateDetails: EmailTemplateDetail[];

	@nested(RolePermission)
	rolePermissions: RolePermission[];

	@nested(ModuleContent)
	moduleContents: ModuleContent[];



}
